### PR TITLE
Add info dump about how to use singletons

### PR DESCRIPTION
**Pull request by prozacchiwawa**
An explanation about how to use the functions in singleton_top_layer and singletons in general lifted from experience using it in checkers.

### CODE REVIEW

File: chia/wallet/puzzles/singleton_top_layer.py [hunk 1]
```diff
@@ -20,6 +20,160 @@
...
+#         |       
+#  ------------------ Atomic Transaction 1 -----------------
+#         v                                          
+#   .------------.       .-------------------------------.
+#   | Launcher   |------>| Eve Coin Containing Program I |
+#   `------------'       `-------------------------------'
+#                                        |                
+#  -------------------- End Transaction 1 ------------------
+#                                        |                
+#  --------------- (2) Transaction With I ------------------
```
Comment (matt-o-how) [inline chia/wallet/puzzles/singleton_top_layer.py:46]: This does not necessarily have to be a new transaction

Comment (prozacchiwawa) [reply chia/wallet/puzzles/singleton_top_layer.py]: Added a comment on that

File: chia/wallet/puzzles/singleton_top_layer.py [hunk 2]
```diff
@@ -20,6 +20,138 @@
...
+# We'll call this puzzle hash (or None) PH.
+#
+#      If this is the Eve singleton:
+#
+#          PH = None
+#          L = LineageProof(Launcher, PH, amount)
+#
+#      Otherwise
+#
+#          PH = ParentOf(S).puzzle_hash
```
Comment (Quexington) [inline chia/wallet/puzzles/singleton_top_layer.py:107]: I think this should be `PH = ParentOf(S).inner_puzzle_hash`

File: chia/wallet/puzzles/singleton_top_layer.py [hunk 3]
```diff
@@ -20,6 +20,138 @@
...
+#
+# Specifically, the required puzzle is the Inner puzzle I for the parent of S
+# unless S is the Eve coin, in which case it is None.
+# So to spend S', the second singleton, I is used, and to spend S'', I' is used.
+# We'll call this puzzle hash (or None) PH.
+#
+#      If this is the Eve singleton:
+#
+#          PH = None
+#          L = LineageProof(Launcher, PH, amount)
```
Comment (Quexington) [inline chia/wallet/puzzles/singleton_top_layer.py:103]: I think in both this line and the corresponding one below, the lineage proof is what you have but plus `.parent_coin_info`.  Remember you're proving the parent's identity so you need to supply the parent's parent when calculating the ID.

**Review by Quexington — DISMISSED** [review]
Some minor comments but otherwise very complete.

**Comment by wjblanke** [comment]
Will wait for Quex's comments. Thanks Arty!

**Comment by wjblanke** [comment]
bump!

File: chia/wallet/puzzles/singleton_top_layer.py (hunk 2 above)
Comment (prozacchiwawa) [reply chia/wallet/puzzles/singleton_top_layer.py:107]: ya.  making this change.

File: chia/wallet/puzzles/singleton_top_layer.py (hunk 3 above)
Comment (prozacchiwawa) [reply chia/wallet/puzzles/singleton_top_layer.py:103]: I think what's being suggested here is that i should point out that the parent_coin_info of the coin being spent according to records we can see via RPC matches the first argument of the LineageProof constructor (Launcher or Parent(S).coin_name).  I've added those remarks.

**Review by wjblanke — APPROVED** [review]
aok

**Pull request merged**